### PR TITLE
linux-yocto: remove deprecated kconfig entries

### PR DESCRIPTION
**Pull request by dancios**
CONFIG_INTEL_MENLOW & CONFIG_EFI_VARS are deprecated

### CODE REVIEW

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

File: meta-dts-distro/recipes-kernel/kernel/linux-yocto/efivars.cfg [hunk 1]
```diff
@@ -1 +0,0 @@
-CONFIG_EFI_VARS=y
```
[the remark this answers is not in the document]
Comment (macpijan) [reply meta-dts-distro/recipes-kernel/kernel/linux-yocto/efivars.cfg:1]: Empty file does not make sense. We should either remove it or replace with the new config.

I have discussed with @dancios and we have agreed that Daniel will provide:
1. Result o checking whether the CONFIG_EFI_VARS is used anywhere in the kernel. If it is not, then it does not do anything anyway, so we can safely remove it.
2. Check if our config perhaps has EFIVAR_FS already enabled.

This explanation should come with this path. I do not see it either here or in the issue: https://github.com/Dasharo/dasharo-issues/issues/833

[the remark this answers is not in the document]
Comment (macpijan) [reply meta-dts-distro/recipes-kernel/kernel/linux-yocto/efivars.cfg:1]: FWIW, we have used efivar sysfs here: https://github.com/Dasharo/dts-scripts/blob/main/scripts/dts-boot#L8

What would be the interface with this new config? Perhaps we shall modify the scripts as well?

**Comment by dancios** [comment]
@macpijan @m-iwanicki Updated PR

**Comment by m-iwanicki** [comment]
@dancios You should answer in review comment if this change fixes it with link do commit/diff. In case of force push you can use link from `Compare` button
image

File: meta-dts-distro/recipes-kernel/kernel/linux-yocto/efivars.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (dancios) [reply meta-dts-distro/recipes-kernel/kernel/linux-yocto/efivars.cfg:1]: @macpijan @m-iwanicki The efivarsfs will be sent in v2. the entries are the same so no changes to apps shold be made.

**Comment by dancios** [comment]
> @dancios You should answer in review comment if this change fixes it with link do commit/diff. In case of force push you can use link from `Compare` button image

Thx for valuable advices I applied suggested changes in previous "comment-thread". They really improve process.

**Pull request closed without merging**